src/platform/ESP32/BLEManagerImpl.h

```diff
@@ -28,9 +28,39 @@
 
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
 
+#include "sdkconfig.h"
+
+#if CONFIG_BT_BLUEDROID_ENABLED
+
 #include "esp_bt.h"
 #include "esp_gap_ble_api.h"
 #include "esp_gatts_api.h"
+#elif CONFIG_BT_NIMBLE_ENABLED
+
+/* min max macros in NimBLE can cause build issues with generic min max
+ * functions defined in CHIP.*/
+#define min
+#define max
+#include "console/console.h"
+#include "esp_nimble_hci.h"
+#include "host/ble_hs.h"
+#include "host/util/util.h"
+#include "nimble/nimble_port.h"
+#include "nimble/nimble_port_freertos.h"
+#include "services/gap/ble_svc_gap.h"
+#undef min
+#undef max
+
```

Comment:
It seems like there is a lot of duplicates #include between this file and nimble/BLEManagerImpl.cpp. Is there any includes that can be removed from this .h file ?

Reply:
Thank you @vivien-apple , removed the unnecessary ones !!